src/test/java/com/limechain/storage/block/tree/BlockTreeTest.java

```diff
@@ -0,0 +1,387 @@
+package com.limechain.storage.block.tree;
+
+import com.limechain.network.protocol.warp.dto.BlockHeader;
+import com.limechain.network.protocol.warp.dto.HeaderDigest;
+import com.limechain.runtime.Runtime;
+import com.limechain.utils.HashUtils;
+import io.emeraldpay.polkaj.types.Hash256;
+import org.apache.tomcat.util.buf.HexUtils;
+import org.javatuples.Pair;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.math.BigInteger;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+
+class BlockTreeTest {
+
+    private static BlockHeader testHeader;
+
+    @BeforeAll
+    public static void init() {
+        testHeader = new BlockHeader();
+        testHeader.setParentHash(getHash("01"));
+        testHeader.setBlockNumber(BigInteger.ZERO);
+        testHeader.setStateRoot(getHash("02"));
+        testHeader.setExtrinsicsRoot(getHash("03"));
+        testHeader.setDigest(new HeaderDigest[0]);
+    }
+
+    @Test
+    void testNewBlockTreeFromNode() {
+        BlockTree bt;
+        List<Node> branches;
+
+        do {
+            Pair<BlockTree, List<Node>> result = createTestBlockTree(testHeader, 5);
+            bt = result.getValue0();
+            branches = result.getValue1();
+
+        } while (branches.isEmpty() || bt.getNode(branches.get(0).getHash()).getChildren().isEmpty());
+
+        Node testNode = bt.getNode(branches.get(0).getHash()).getChildren().get(0);
+        List<Node> leaves = testNode.getLeaves();
+
+//        BlockTree newBt = new BlockTree(testNode);
+//        assertEquals(new HashSet<>(leaves), new HashSet<>(newBt.getLeaves().nodes()));
```

Comment:
No point in this test being ran. Either uncomment the assertions or delete the whole test case.